mock/py/mockbuild/plugins/rpkg_preprocessor.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+# vim:expandtab:autoindent:tabstop=4:shiftwidth=4:filetype=python:textwidth=0:
+# License: GPL2 or later see COPYING
+# Written by clime
+# Copyright (C) 2020 clime <clime@fedoraproject.org>
+
+# python library imports
+import os
+import os.path
+import re
+import subprocess
+import configparser
+import shlex
+
+# our imports
+import mockbuild.util
+from mockbuild.trace_decorator import getLog, traceLog
+from mockbuild.exception import PkgError
+
+requires_api_version = "1.1"
+
+
+# plugin entry point
+@traceLog()
+def init(plugins, conf, buildroot):
+    RpkgPreprocessor(plugins, conf, buildroot)
+
+
+class RpkgPreprocessor(object):
+    """preprocess spec file by using rpkg utilities"""
+    # pylint: disable=too-few-public-methods
+    @traceLog()
+    def __init__(self, plugins, conf, buildroot):
+        self.buildroot = buildroot
+        self.config = buildroot.config
+        self.state = buildroot.state
+        self.opts = conf
+        self.log = getLog()
+        plugins.add_hook("pre_srpm_build", self._preprocess_proxy)
+        self.log.info("rpkg_preprocessor: initialized")
+
+    @traceLog()
+    def _install_requires(self, requires):
+        try:
+            self.buildroot.uid_manager.becomeUser(0, 0)
+            self.buildroot.pkg_manager.install(*requires, check=True)
+        finally:
+            self.buildroot.uid_manager.restorePrivs()
```

Comment:
Please take a look at `buildroot.preexisting_deps`.  Adding the packages there should be enough.

Reply:
It doesn't seem to work for me. I tried to follow examples of other plugins like `ccache`. I looked at the code paths and it seems the `buildroot.preexisting_deps` is only used from `installSrpmDeps`, which doesn't seem to be invoked for `backend.py:buildsrpm` method.

Reply:
Indeed, that seems to be a bug.  We should call it on universal place...
The `ccache` is not srpm dependency.  Do you want me to fix this first,
or can you write a patch also for this?  The patch should be applied before
this patch.

Reply:
I would prefer to finish this pr in some working state and perhaps do some optimizations later.